airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeV1V2Migrator.kt

```diff
@@ -59,45 +76,55 @@ class SnowflakeV1V2Migrator(
         // The obvious database.getMetaData().getColumns() solution doesn't work, because JDBC
         // translates
         // VARIANT as VARCHAR
-        val columns =
-            database
-                .queryJsons(
+
+        var showColumnsResult : List<JsonNode> = listOf()
+
+        try {
+            val showColumnsQuery =
+                String.format(
                     """
-            SELECT column_name, data_type, is_nullable
-            FROM information_schema.columns
-            WHERE table_catalog = ?
-              AND table_schema = ?
-              AND table_name = ?
-            ORDER BY ordinal_position;
-            
-            """.trimIndent(),
+                       SHOW COLUMNS IN TABLE "%s"."%s"."%s";
+                    """.trimIndent(),
                     databaseName,
-                    namespace!!,
-                    tableName!!
+                    namespace,
+                    tableName,
                 )
+            showColumnsResult = database.queryJsons(
+                showColumnsQuery
+            )
+            val columnsFromShowQuery = showColumnsResult
                 .stream()
                 .collect(
                     { LinkedHashMap() },
                     { map: java.util.LinkedHashMap<String, ColumnDefinition>, row: JsonNode ->
-                        map[row["COLUMN_NAME"].asText()] =
+                        map[row["column_name"].asText()] =
                             ColumnDefinition(
-                                row["COLUMN_NAME"].asText(),
-                                row["DATA_TYPE"].asText(),
+                                row["column_name"].asText(),
+                                //row["data_type"].asText(),
+                                JSONObject(row["data_type"].asText()).getString("type"),
```

Comment:
is there a difference introduced by the move to SHOW COLUMNS (similar to the call above)?

Reply:
yes, there are differences in the output of the information_schema query and SHOW query